lib/converters/html-converter.js

```diff
@@ -102,4 +106,30 @@ module.exports = class HtmlConverter extends Converter {
       }
     });
   }
+
+  /**
+   * This is a wrapper around convert(), that will render some default HTML if the expected conversion fails.
+   * Useful in error conditions.
+   *
+   * @param htmlInfo - see convert()
+   * @param data - see convert()
+   *
+   * @param message - {string} text to display inside a paragraph of the default HTML, if data cannot be converted as expected
+   * @param logger - optional logger instance, logs conversion error that causes default HTML rendering
+   *
+   * @returns - see convert()
+   */
+  static async convertWithDefault(htmlInfo, data, message, logger) {
```

Comment:
I explained in the modified comment header of the function:
```
   * Call this function in stead of convert(), to complete requests in error conditions.
```
I called `HTMLConverter.convertWithDefault()` in all those cases (see handlers/response-handler.js and handlers/content-negotiation-handler.js), in stead of `HTMLConverter.convertWithDefault()`.

A simple test:
- modify example/config-errors.yaml to not contain 404, 500 pages
- browse http://localhost:3000/thisPageSurelyWontExist 
- browse http://localhost:3000/bad_pipeModule

Earlier code will result in no response being returned in both cases

Reply:
Why not just send the status code instead of embedding HTML here if there is no HTML specified by the user?
Ideally, all HTML is defined by the user.

Reply:
Can be done. I see two possible solutions:
1) remove the parameter message and return empty string in case of error (requires rename of the function from convertWithDefault to something like convertForced). This would be my choice.
2) remove the method entirely and restore the code that calls convertWithDefault() to call convert() again as before, and add a catch block that sets html to empty string (and optionally does a logger.error() call.

What's your preference?

Reply:
I would go for option 2 and mention the problem and your solution in a separate issue, so we can keep track if there.

Reply:
OK, I'll do it like that.

Reply:
Done